### DISCUSSION TITLE
Mention does not work properly when html tag is put before mention prefix.

### DISCUSSION THREAD

**Discussion by wadakatu** [thephpleague/commonmark#879] [Q&A] [answered]
Mention extension does not work when I put html tag before mention prefix which is '@' .
Seemingly, there is no `<a>` tag in the converted result.

## configuration
```
 $environment = new Environment([
                'html_input' => 'escape',
                'mentions' => [
                    'username' => [
                        'prefix' => '@',
                        'pattern' => '[a-z\d](?:[a-z\d]|-(?=[a-z\d])){0,38}(?!\w)',
                        'generator' => config('app.url').'/user-info/%s',
                    ],
                ],
            ]);

            $environment->addExtension(new CommonMarkCoreExtension);
            $environment->addExtension(new MentionExtension);

            return new LeagueConverter(new MarkdownConverter($environment));
```

## example markdown
```
<p>test</p> @wadakatu
```

## expected  output
```
<p>&lt;p&gt;test&lt;/p&gt; <a href="https://test/user-info/wadakatu">@wadakatu</a></p>
```

## actual output (no `<a>` tag.)
```
&lt;p&gt;test&lt;/p&gt; @wadakatu
```

Could you tell me whether or not this behavior is intentional?
Is there any way to accomplish expected output ?

Thank you.

**Comment by colinodell** [comment, marked as answer]
Hi there!

According to [the CommonMark spec](https://spec.commonmark.org/0.30/#html-blocks), your example Markdown input contains an HTML block (type 6) which includes `@wadakatu` as-is. With HTML blocks:

> Everything until the next blank line or end of document gets included in the HTML block. So, in the [...] example, what looks like a Markdown code block is actually part of the HTML block, which continues until a blank line or the end of the document is reached

The behavior you're seeing will happen with any type of Markdown that co-exists on the same line as the HTML, like [in these examples](https://babelmark.github.io/?text=%3Cp%3Etest%3C%2Fp%3E+**sdfsdf**%0A%0A%3Cp%3Etest%3C%2Fp%3E+%5Bmarkdown+is+fun!%5D(www.example.com)%0A%0A%3Cp%3Etest%3C%2Fp%3E+%23+Test).

To accomplish the expected output, you'll need to move the mention out to a separate line like this:

```md
<p>test</p>

@wadakatu
```

Or use an [Raw HTML inline](https://spec.commonmark.org/0.30/#raw-html) (instead of an HTML block) by starting the line with some non-HTML like this:

```md
This is a <p>test</p> @wadakatu
```

**Reply by wadakatu** [reply]
@colinodell 
Thank you for your prompt answer.
I think that your first suggestion is suitable for me.

Best Regards.
